mlir/lib/Dialect/Arith/IR/ArithCanonicalization.td

```diff
@@ -403,7 +405,7 @@ def TruncIShrSIToTrunciShrUI :
 def TruncIShrUIMulIToMulSIExtended :
     Pat<(Arith_TruncIOp:$tr (Arith_ShRUIOp
                               (Arith_MulIOp:$mul
-                                (Arith_ExtSIOp $x), (Arith_ExtSIOp $y)),
+                                (Arith_ExtSIOp $x), (Arith_ExtSIOp $y), $ovf1),
```

Comment:
If we known that the multiplication doesn't overflow we should be able to replace the overflow result bit with 0.

Could you add a top-level TODO comment to audit these canon patterns with overflow flags in mind? I think this would make a nice self-contained issue we could mark as 'good first issue'